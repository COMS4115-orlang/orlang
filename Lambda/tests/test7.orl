(* unification error: main must be of type int *)

val operator_if : Bool -> 'aaa -> 'aaa -> 'bbb
let operator_if c t e = _if c _then t _else e

let main = (\x -> x + 1)
