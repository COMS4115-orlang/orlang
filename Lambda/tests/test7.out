Fatal error: exception Failure("Cannot unify arrow type (Int) -> (Int) with unit type")
